src/commands/createStaticWebApp/createStaticWebApp.ts

```diff
@@ -143,29 +148,195 @@ export async function createStaticWebApp(context: IActionContext & Partial<ICrea
+    const swa: StaticSiteARMResource = nonNullProp(wizardContext, 'staticWebApp');
+    await gitPull(nonNullProp(wizardContext, 'repo'));
+    const appResource: AppResource = {
+        id: nonNullProp(swa, 'id'),
+        name: nonNullProp(swa, 'name'),
+        type: nonNullProp(swa, 'type'),
+        ...swa
+    };
+    const resolver = new StaticWebAppResolver();
+    const resolvedSwa = await resolver.resolveResource(node.subscription, appResource);
+    if (resolvedSwa) {
+        await showSwaCreated(resolvedSwa);
+    }
+    return appResource;
+}
+
+export async function createStaticWebAppWithLogicApp(context: IActionContext & Partial<ICreateChildImplContext> & Partial<IStaticWebAppWizardContext>, node?: SubscriptionTreeItemBase): Promise<AppResource> {
+    if (isVerifyingWorkspace) {
+        throw new VerifyingWorkspaceError(context);
+    }
+    const progressOptions: ProgressOptions = {
+        location: ProgressLocation.Window,
+        title: localize('verifyingWorkspace', 'Verifying workspace...')
+    };
+
+    isVerifyingWorkspace = true;
+    if (!isSubscription(node)) {
+        node = await ext.rgApi.appResourceTree.showTreeItemPicker<SubscriptionTreeItemBase>(SubscriptionTreeItemBase.contextValue, context);
+    }
+    const client: WebSiteManagementClient = await createWebSiteClient([context, node.subscription]);
+    const wizardContext: IStaticWebAppWizardContext = {
+        accessToken: await getGitHubAccessToken(),
+        client,
+        ...context,
+        ...node.subscription,
+        ...(await createActivityContext())
+    };
+    const title: string = localize('createStaticApp', 'Create Static Web App');
+    const promptSteps: AzureWizardPromptStep<IStaticWebAppWizardContext & ExecuteActivityContext>[] = [];
+    const executeSteps: AzureWizardExecuteStep<IStaticWebAppWizardContext>[] = [];
+    if (!context.advancedCreation) {
+        //[1] gets standard and not free
+        wizardContext.sku = SkuListStep.getSkus()[1];
+        executeSteps.push(new ResourceGroupCreateStep());
+    } else {
+        promptSteps.push(new ResourceGroupListStep());
+    }
 
+    promptSteps.push(new StaticWebAppNameStep(), new SkuListStep());
     const wizard: AzureWizard<IStaticWebAppWizardContext> = new AzureWizard(wizardContext, {
         title,
         promptSteps,
         executeSteps,
         showLoadingPrompt: true
     });
 
-    wizardContext.telemetry.properties.gotRemote = String(hasRemote);
     wizardContext.uri = wizardContext.uri || getSingleRootFsPath();
+    await wizard.prompt();
+
+    //create github template
+    const folderName: string = wizardContext.newStaticWebAppName || 'default_folder_name';
+    const homeDir = homedir();
+    const baseClonePath = join(homeDir, folderName);
+    const clonePath = join(baseClonePath, 'static-web-app');
+    const clonePathUri: Uri = Uri.file(clonePath);
+    const octokitClient: Octokit = await createOctokitClient(context);
```

Comment:
This code might be hard to interpret for someone not having the full context. Do you think maybe breaking it into smaller pieces can help with readability. 
Something like 

        const folderName = getFolderName(wizardContext);
        const clonePath = getClonePath(folderName);
        const clonePathUri: Uri = Uri.file(clonePath);
        const octokitClient: Octokit = await createOctokitClient(context);
        const response = await createRepository(octokitClient, folderName);

Reply:
And same below